sql/core/src/main/scala/org/apache/spark/sql/internal/HiveSerDe.scala

```diff
@@ -77,4 +79,16 @@ object HiveSerDe {
 
     serdeMap.get(key)
   }
+
+  def getDefaultStorage(conf: SQLConf): CatalogStorageFormat = {
+    val defaultStorageType = conf.getConfString("hive.default.fileformat", "textfile")
+    val defaultHiveSerde = sourceToSerDe(defaultStorageType)
+    CatalogStorageFormat.empty.copy(
+      inputFormat = defaultHiveSerde.flatMap(_.inputFormat)
+        .orElse(Some("org.apache.hadoop.mapred.TextInputFormat")),
+      outputFormat = defaultHiveSerde.flatMap(_.outputFormat)
+        .orElse(Some("org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat")),
+      serde = defaultHiveSerde.flatMap(_.serde)
+        .orElse(Some("org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe")))
```

Comment:
it's a little different from what it was. Previously we don't set default serde at [parser](https://github.com/apache/spark/pull/16296/files#diff-1bb4f7bd5a2656f48bcd3c857167a11bL1069), but set it during [analysis](https://github.com/apache/spark/pull/16296/files#diff-c4ed9859978dd6ac271b6a40ee945e4bL54). However, this doesn't make sense and I think we should just provide the default serde at the beginning.